### DISCUSSION TITLE
Upgrade process for distributed cluster?

### DISCUSSION THREAD

**Discussion by Plloi** [seaweedfs/seaweedfs#3343] [Q&A] [answered]
So i currently have 3 masters, 2 filers, 6 volume servers, and 3 fuse clients. with expectations that i'll be adding more volume servers and clients.

Is a version update a downtime event? or can I do a rolling update across my nodes?

If a rolling update is possible, is there recommended order for rolling out?

**Comment by chrislusf** [comment, marked as answer]
Rolling restart from masters, volume servers, filers, mount, s3, etc, in this order. Basically from lower core layers to upper layers.

**Reply by Plloi** [reply]
As I suspected but couldn't find in the docs, thank you.

**Reply by meotimdihia** [reply]
@chrislusf we should have an upgrade guide. 
Currently, I am using `seaweedfs` for image storage. And I just have 2 images server. The total size is 40TB.
When I upgrade seaweeds, it is always down for 15 minutes -> half an hour.
I wonder if we have any way to upgrade without downtime. And what are the requirements?

I think we have many people like me that are not experts in this problem.

**Reply by gaby** [reply]
I'm also interested in knowing how to upgrade with big amounts of data. Specially if running with Docker

**Reply by meotimdihia** [reply]
I created a new discusstion: https://github.com/seaweedfs/seaweedfs/discussions/3800

**Reply by gaby** [reply]
@meotimdihia Thank you!
